lib/cpp/src/thrift/transport/THeaderTransport.cpp

```diff
@@ -375,7 +383,12 @@ void THeaderTransport::resetProtocol() {
 }
 
 uint32_t THeaderTransport::getWriteBytes() {
-  return wBase_ - wBuf_.get();
+  ptrdiff_t wb = wBase_ - wBuf_.get();
+  if (wb > std::numeric_limits<uint32_t>().max()) {
+    throw TTransportException(TTransportException::CORRUPTED_DATA,
+                              "write size is unreasonable");
+  }
```

Comment:
as before - code repetition - some utility function would make it much cleaner (different than before, as now TTTransportException should be thrown)